src/policy_evaluator.rs

```diff
@@ -32,19 +33,26 @@ impl fmt::Display for PolicyExecutionMode {
     }
 }
 
-#[derive(Debug, Serialize)]
-pub struct ValidateRequest(pub(crate) serde_json::Value);
+/// A validation request that can be sent to a policy evaluator.
+/// It can be either a raw JSON object, or a Kubernetes AdmissionRequest.
+#[derive(Clone, Debug, Serialize)]
+#[serde(untagged)]
+pub enum ValidateRequest {
+    Raw(serde_json::Value),
+    AdmissionRequest(AdmissionRequest),
+}
 
 impl ValidateRequest {
-    pub fn new(request: serde_json::Value) -> Self {
-        ValidateRequest(request)
-    }
-
     pub(crate) fn uid(&self) -> &str {
-        if let Some(uid) = self.0.get("uid").and_then(value::Value::as_str) {
-            uid
-        } else {
-            ""
+        match self {
+            ValidateRequest::Raw(raw_req) => {
+                if let Some(uid) = raw_req.get("uid").and_then(value::Value::as_str) {
+                    uid
+                } else {
+                    ""
+                }
```

Comment:
IMHO this looks better:
```suggestion
            ValidateRequest::Raw(raw_req) => raw_req
                .get("uid")
                .and_then(value::Value::as_str)
                .unwrap_or_default(),
```